src/Symfony/Component/HttpClient/NoPrivateNetworkHttpClient.php

```diff
@@ -74,7 +59,7 @@ public function request(string $method, string $url, array $options = []): Respo
 
         $options['on_progress'] = function (int $dlNow, int $dlSize, array $info) use ($onProgress, $subnets, &$lastPrimaryIp): void {
             if ($info['primary_ip'] !== $lastPrimaryIp) {
-                if ($info['primary_ip'] && IpUtils::checkIp($info['primary_ip'], $subnets ?? self::PRIVATE_SUBNETS)) {
+                if ($info['primary_ip'] && IpUtils::checkIp($info['primary_ip'], $subnets ?? IpUtils::PRIVATE_SUBNETS)) {
```

Comment:
This code will require `symfony/http-foundation > 6.3-dev`. The composer.json of `http-client` does not mention `http-foundation`, a conflict entry may be necessary. WDYT?

Reply:
Good catch, but this issue has already been addressed. To use the `NoPrivateNetworkHttpClient`, http-foundation is a required dependency.

https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/HttpClient/NoPrivateNetworkHttpClient.php#L57-L59

Reply:
My bad, missed the part about the conflict. Yeah, that would be an issue.

Reply:
yes